crates/knit-wit/src/lib.rs

```diff
@@ -0,0 +1,171 @@
+use anyhow::anyhow;
+use std::{
+    collections::HashMap,
+    path::{Path, PathBuf},
+};
+use wit_component::{DecodedWasm, WitPrinter};
+use wit_parser::{PackageId, Resolve};
+
+wit_bindgen::generate!({
+    world: "knit-wit",
+});
+
+struct KnitWit;
+
+impl Guest for KnitWit {
+    fn knit_wit(
+        wit_paths: Vec<String>,
+        worlds: Vec<String>,
+        output_world: Option<String>,
+        output_package: Option<String>,
+        output_dir: Option<String>,
+    ) -> Result<Vec<(String, String)>, String> {
+        if wit_paths.is_empty() {
+            return Err("No wit-paths provided".to_owned());
+        }
+        if worlds.is_empty() {
+            return Err("No worlds provided to be knit".to_owned());
+        }
+
+        let output_package = output_package.unwrap_or("local:combined-wit".to_owned());
+        let output_world = output_world.unwrap_or("combined".to_owned());
+        let output_dir = output_dir.unwrap_or("combined-wit".to_owned());
```

Comment:
```suggestion
        let output_package = output_package.unwrap_or_else(|| "local:combined-wit".to_owned());
        let output_world = output_world.unwrap_or_else(|| "combined".to_owned());
        let output_dir = output_dir.unwrap_or_else(|| "combined-wit".to_owned());
```

Reply:
I am curious to understand the difference here.

Reply:
It's just a nit. The previous version always allocates the default strings on the heap (which is a waste if you don't use them). This version only turns the `&str` into `String`s if they are used. At the end of the day this makes very little difference but it's just good practice.